aptos-move/move-examples/token_migration/sources/token_migration.move

```diff
@@ -0,0 +1,116 @@
+    use aptos_token_objects::token as base_tokenv2;
+    use aptos_token_objects::royalty as tokenv2_royalty;
+    use std::option::{Self, Option};
+    use std::string::String;
+    use std::signer;
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Storage state for managing the no-code Token.
+    struct MigratedToken has key {
+        /// Token to wrap
+        token: tokenv1::Token,
+        /// Used to burn.
+        burn_ref: Option<base_tokenv2::BurnRef>,
+        /// Used to control freeze.
+        transfer_ref: Option<object::TransferRef>,
+        /// Used to mutate fields
+        mutator_ref: Option<base_tokenv2::MutatorRef>,
+        /// Used to mutate fields
+        extend_ref: Option<ExtendRef>,
+    }
+
+    /// Create a collection based on the previous collection
+    public entry fun create_migrated_collection(creator: &signer, collection_name: String) {
+        let creator_address = signer::address_of(creator);
+
+        let collection_description = tokenv1::get_collection_description(creator_address, collection_name);
+        let collection_uri = tokenv1::get_collection_uri(creator_address, collection_name);
+        let max_supply = tokenv1::get_collection_maximum(creator_address, collection_name);
+        let collection_mutability_config = tokenv1::get_collection_mutability_config(creator_address, collection_name);
+
+        // For now, allow mutable everything, TODO: Make this configurable / use previous state
+        tokenv2::create_collection(
+            creator,
+            collection_description,
+            max_supply,
+            collection_name,
+            collection_uri,
+            tokenv1::get_collection_mutability_description(&collection_mutability_config),
+            true,
+            tokenv1::get_collection_mutability_uri(&collection_mutability_config),
+            // In v1, token mutability configs is set on the token instead of collection.
+            // This example defaults to true here but the creator can deploy custom code built on token v2 to maintain
+            // the same control setup as token v1 if desired.
+            true, /* mutable_token_description */
+            true, /* mutable_token_name */
+            true, /* mutable_token_properties */
+            true, /* mutable_token_uri */
+            true, /* tokens_burnable_by_creator */
+            true, /* tokens_freezable_by_creator */
+            // Royalty is set on the tokens in v1, so we'll maintain the same setup during migration. This sets
+            // the royalty on collections, so we'll set it to 0.
+            0,
+            1
+        );
+    }
+
+    /// This can only called by the creator, who also needs to own the token being migrated.
+    /// If creators want to let the token owner migrate their own tokens, they would need to deploy custom code.
+    public fun migrate_token(creator: &signer, receiver: address, collection_name: String, token_name: String) {
```

Comment:
Why does this need to be called by the creator? Feels like it should be callable by anyone that has the token. As an entry fun could be callable by whomever has it in their TokenStore?

Reply:
So, it looks like the state that this was left in is broken and is half migrated to `AptosToken` which doesn't work with the token level royalties.

This will need to be rewritten over again with the standard `TokenObject`